src/main/java/org/gridsuite/network/map/dto/voltagelevel/VoltageLevelMapInfos.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.network.map.dto.voltagelevel;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.VoltageLevel;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+@SuperBuilder
+@Getter
+public class VoltageLevelMapInfos extends AbstractVoltageLevelInfos {
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    private String substationId;
+
+    private double nominalVoltage;
+
+    public static VoltageLevelMapInfos toData(Identifiable<?> identifiable) {
+        VoltageLevel voltageLevel = (VoltageLevel) identifiable;
+        return VoltageLevelMapInfos.builder()
+                .id(voltageLevel.getId())
+                .name(voltageLevel.getOptionalName().orElse(null))
+                .substationId(voltageLevel.getSubstation().orElseThrow().getId())
```

Comment:
if voltageLevel has no substation, we should no throw an exception ...

Reply:
Same remark as in my previous response